src/poetry/utils/env.py

```diff
@@ -812,6 +823,13 @@ def create_venv(
 
         if env.is_venv() and not force:
             # Already inside a virtualenv.
+            current_python = Version.parse(
+                ".".join(str(c) for c in env.version_info[:3])
+            )
+            if not self._poetry.package.python_constraint.allows(current_python):
+                raise InvalidCurrentPythonVersionError(
+                    self._poetry.package.python_versions, current_python
```

Comment:
`current_python` is of type `Version`, but the type hint in `InvalidCurrentPythonVersionError` says it should be a string. Thus, either the type hint has to be changed or `current_python` should be cast to string explicity `str(current_python)`. I tend toward the latter.

Reply:
that's true.

```suggestion
                    self._poetry.package.python_versions, str(current_python)
```